File: LeetCode_Solutions/LeetCode_Solutions/RemoveDuplicatesFromSortedList_83.m
//
//  RemoveDuplicatesFromSortedList_83.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/11/10.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//




#import "RemoveDuplicatesFromSortedList_83.h"

@implementation RemoveDuplicatesFromSortedList_83


- (struct ListNode *)deleteDuplication:(struct ListNode*)head {

    if ( head->next == nil) {
        return head;
    }
    while (head) {
        
        if (head->val == head->next->val) {
            [self deleteList:head->next];
        } else {
            head = head -> next;
        }
    }
    return head;
}

- (struct ListNode *)deleteList:(struct ListNode*)node {

    node->val = node->next->val;
    node->next = node->next->next;
    
    return node;
}

@end

File: LeetCode_Swift/LeetCode_Swift/RemoveDuplicatesFromSortedList_83.swift
//
//  RemoveDuplicatesFromSortedList_83.swift
//  LeetCode_Swift
//
//  Created by Huni on 21/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class RemoveDuplicatesFromSortedList_83 {

    func deleteDuplicates(_ head: ListNode?) -> ListNode? {
       
        guard let head = head else {
            return nil
        }
        var curt = head
        while (curt.next != nil) {
            if curt.val == curt.next?.val {
                self.removeList(curt.next)
            } else {
                curt = curt.next!
            }
        }
        
        return head
    }
    
    func removeList(_ node:ListNode?) {
        node?.next = node?.next?.next
        node?.val = (node?.next?.val)!
    }
    
}

